prime-dotnet-webapi/Services/HealthAuthorityService.cs

```diff
@@ -76,23 +77,37 @@ public async Task<bool> AuthorizedUserExistsOnHealthAuthorityAsync(int healthAut
                 .AnyAsync();
         }
 
-        public async Task UpdateCareTypesAsync(int healthAuthorityId, IEnumerable<string> careTypes)
+        public async Task<bool> UpdateCareTypesAsync(int healthAuthorityId, IEnumerable<string> careTypes)
```

Comment:
We should add a check either in this method or in the controller to prevent adding two identical care types, since it breaks the Entity Matcher

Reply:
The same check should be done on a vendor as well right?

Reply:
For vendors I created a method ```AnyVendorExistsOnHealthAuthorityAsync``` that takes in a health authority id and  an array of vendor codes and checks if they already exist within the health authority. Alternatively we could do a `foreach` and use the existing ```VendorExistsOnHealthAuthorityAsync``` that takes in one vendor code instead of a list.

Reply:
Done